File: src/main/java/pl/yalgrin/playnite/simplesync/config/JsonConfig.java
package pl.yalgrin.playnite.simplesync.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Primary;
import tools.jackson.databind.json.JsonMapper;

@Configuration
public class JsonConfig {
    @Primary
    @Bean
    public JsonMapper jsonMapper() {
        JsonMapper.Builder builder = JsonMapper.builder();
        JsonMapperProvider.customizer().customize(builder);
        return builder.build();
    }
}

File: src/main/kotlin/pl/yalgrin/playnite/simplesync/common/config/JsonConfig.kt
package pl.yalgrin.playnite.simplesync.common.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Primary
import tools.jackson.databind.json.JsonMapper

@Configuration
class JsonConfig {
    @Primary
    @Bean
    fun jsonMapper(): JsonMapper {
        return buildJsonMapper()
    }
}